### DISCUSSION TITLE
Modify babel config as part of our plugin on file base

### DISCUSSION THREAD

**Discussion by fkrauthan** [babel/babel#15084] [Q&A] [answered]
I have a somewhat unique problem. I want to transpile a bunch of files through babel. My problem is: Depending on the first comment header I need different targets (e.g. some files might be targeting the browser and some targeting node). Is there any way to write a plugin that gets invoked first. For example, do a simple string contains check to see what type of file it is and then modify the targets or do I have to try to cover everything based on include and exclude patterns and based on filenames? Or is there maybe even a third option?

**Comment by JLHwung** [comment, marked as answer]
A plugin could manipulate Babel options by `manipulateOptions()`, but it could do so only at the config resolving phase. At the time when Babel is processing the input source, the config can no longer be mutated.

> cover everything based on include and exclude patterns and based on filenames

Yes. And you can read all sources, check certain strings, and programmatically generate `input` and `exclude` for `babel.config.js`.

**Reply by fkrauthan** [reply]
Ok so pretty much my only option is calling babel multiple times with different input, exclude settings. Or is there a way to have overwrite settings (similar to what [eslint](https://eslint.org/docs/latest/user-guide/configuring/rules#using-configuration-files-1) offers) to change settings for a couple of files but not all of them?

**Reply by JLHwung** [reply]
You can use the [`overrides` option](https://babeljs.io/docs/en/options#overrides).

**Reply by fkrauthan** [reply]
Oh nice so I could combine overrides with input, exclude rules to have a single babel config and a single babel cli command to apply different targets for example? That's pretty cool thank you/
